java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -69,14 +70,10 @@ public UnleashEngine(List<IStrategy> customStrategies, IStrategy fallbackStrateg
         }
     }
 
-    public void takeState(String toggles) throws YggdrasilInvalidInputException {
-        YggResponse<Void> response = read(yggdrasil.takeState(this.enginePtr, toggles),
+    public void takeState(String toggles) throws YggdrasilInvalidInputException, YggdrasilError {
```

Comment:
All the changes from here to line 148 are basically the same. Use read now directly returns the object requested rather than the Yggdrasil wrapping response. If the method passes a string to Yggdrasil, that's now encoded explicitly as UTF-8 before send